File: frontend/src/types/sensor.ts
export interface SensorDevice {
  id: number;
  building_id: number;
  floor_id: number | null;
  room_name: string | null;
  device_id: string;
  name: string;
  sensor_type: 'mq2' | 'temp';
  threshold: number;
  latest_value: number;
  latest_status: 'safe' | 'danger';
  unit: string;
  last_seen_at: string | null;
}

export interface DashboardSummary {
  total_mq2: number;
  total_temperature: number;
  safe_count: number;
  danger_count: number;
  latest_updated_at: string | null;
}

File: backend/app/schemas/sensor.py
from datetime import datetime
from pydantic import BaseModel, ConfigDict


class SensorDeviceResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    building_id: int
    floor_id: int | None
    room_name: str | None
    device_id: str
    name: str
    sensor_type: str
    threshold: float
    latest_value: float
    latest_status: str
    unit: str
    last_seen_at: datetime | None


class SensorReadingResponse(BaseModel):
    id: int
    device_id: str
    name: str
    sensor_type: str
    value: float
    status: str
    unit: str
    created_at: datetime
    floor_id: int | None
    room_name: str | None


class DashboardSummaryResponse(BaseModel):
    total_mq2: int
    total_temperature: int
    safe_count: int
    danger_count: int
    latest_updated_at: datetime | None


class DeviceIngestRequest(BaseModel):
    buildingCode: str
    deviceId: str
    sensorType: str
    value: float
    unit: str
    timestamp: datetime | None = None
